VictoriaAlbert/VictoriaAlbert/MuseumTableViewController.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  MuseumTableViewController.swift
+//  VictoriaAlbert
+//
+//  Created by Harichandan Singh on 11/10/16.
+//  Copyright © 2016 Harichandan Singh. All rights reserved.
+//
+
+import UIKit
+
+class MuseumTableViewController: UITableViewController, UISearchBarDelegate {
+    //MARK: Properties
+    var searchValue: String = "ring"
+    var apiEndpoint: String {
+        return "http://www.vam.ac.uk/api/json/museumobject/search?q=\(searchValue)"
+    }
+    internal var museums: [Museum] = []
+    
+    //MARK: - Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loadMuseum()
+        createSearchBar()
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    func loadMuseum() {
+        APIRequestManager.manager.getData(apiEndpoint: apiEndpoint) { (data: Data) in
+            if let museums = Museum.turnDataIntoMuseumArray(data: data) {
+                self.museums = museums
+            }
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    func createSearchBar() {
+        let searchBar = UISearchBar()
+        searchBar.placeholder = "Search for an object in the Victoria and Albert museum."
+        searchBar.delegate = self
+        self.navigationItem.titleView = searchBar
+    }
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let text = searchBar.text else { return }
+        searchValue = text
+        APIRequestManager.manager.getData(apiEndpoint: apiEndpoint) { (data: Data) in
+            if let museums = Museum.turnDataIntoMuseumArray(data: data) {
+                self.museums = museums
+            }
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    // MARK: - Table view data source
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return self.museums.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "museumCell", for: indexPath)
+        let museumItem = museums[indexPath.row]
+        
+        //image URL string construction
+        let imageID = museumItem.primaryImageID
+        let firstSixOfImageID = imageID[imageID.startIndex..<imageID.index(imageID.startIndex, offsetBy: 6)]
```

Comment:
This line crashes when there is no primaryImageID. You could check for an empty string (which you're initializing to) or could have made that field optional.